src/main/java/frc/bionic/swerve/SwerveModule.java

```diff
@@ -0,0 +1,260 @@
+/*
+ * Team 4909, Bionics
+ * Billerica Memorial High School
+ *
+ * Copyright:
+ *   2021 Bionics
+ *
+ * License:
+ *   MIT: https://opensource.org/licenses/MIT
+ *   See the LICENSE file in the project's top-level directory for details.
+ */
+
+package frc.bionic.swerve;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+
+
+public class SwerveModule
+{
+  // Gear ratio of first two pairs of gears.
+  // Yaw does not depend on the third pair.
+  private double             GEAR_RATIO_12;
```

Comment:
we may want to think of a better name for this as Matt's swerve only has a single reduction before the differential. And actually has a reduction after the differential.

Reply:
Ok, I'll correct that.